pkg/apiserver/watch_test.go

```diff
@@ -111,65 +137,159 @@ func TestWatchWebsocket(t *testing.T) {
 	}
 }
 
-func TestWatchHTTP(t *testing.T) {
+func TestWatchRead(t *testing.T) {
 	simpleStorage := &SimpleRESTStorage{}
+	_ = rest.Watcher(simpleStorage) // Give compile error if this doesn't work.
 	handler := handle(map[string]rest.Storage{"simples": simpleStorage})
 	server := httptest.NewServer(handler)
 	// TODO: Uncomment when fix #19254
 	// defer server.Close()
-	client := http.Client{}
 
 	dest, _ := url.Parse(server.URL)
-	dest.Path = "/" + prefix + "/" + testGroupVersion.Group + "/" + testGroupVersion.Version + "/watch/simples"
-	dest.RawQuery = ""
+	dest.Path = "/" + prefix + "/" + testGroupVersion.Group + "/" + testGroupVersion.Version + "/simples"
+	dest.RawQuery = "watch=1"
 
-	request, err := http.NewRequest("GET", dest.String(), nil)
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
+	connectHTTP := func(accept string) (io.ReadCloser, string) {
+		client := http.Client{}
+		request, err := http.NewRequest("GET", dest.String(), nil)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		request.Header.Add("Accept", accept)
 
-	response, err := client.Do(request)
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
+		response, err := client.Do(request)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
```

Comment:
I mean, that this is body connectHTTP function - so it should should indeted by one more tab, similarly the the above lines.